vms/evm/uptimetracker/state.go

```diff
@@ -0,0 +1,266 @@
+// Copyright (C) 2019-2025, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+
+package uptimetracker
+
+import (
+	"fmt"
+	"math"
+	"time"
+
+	"github.com/ava-labs/avalanchego/codec"
+	"github.com/ava-labs/avalanchego/database"
+	"github.com/ava-labs/avalanchego/ids"
+	"github.com/ava-labs/avalanchego/snow/uptime"
+	"golang.org/x/exp/maps"
+	"github.com/ava-labs/avalanchego/codec/linearcodec"
+)
+
+const (
+	codecVersion  uint16         = 0
+	updatedStatus dbUpdateStatus = iota
+	deletedStatus
+)
+
+var (
+	codecManager codec.Manager
+	_            uptime.State = (*state)(nil)
+)
+
+func init() {
+	codecManager = codec.NewManager(math.MaxInt32)
+	c := linearcodec.NewDefault()
+
+	if err := c.RegisterType(validator{}); err != nil {
+		panic(fmt.Errorf("failed to register type: %w", err))
+	}
+
+	if err := codecManager.RegisterCodec(codecVersion, c); err != nil {
+		panic(fmt.Errorf("failed to register codec: %w", err))
+	}
+}
+
+type dbUpdateStatus int
+
+type validator struct {
+	UpDuration    time.Duration `serialize:"true"`
+	LastUpdated   uint64        `serialize:"true"`
+	NodeID        ids.NodeID    `serialize:"true"`
+	Weight        uint64        `serialize:"true"`
+	StartTime     uint64        `serialize:"true"`
```

Comment:
We have these fields because we used to use this to serve the `GetCurrentValidators` api on the evm, but since the new api for this package just cares about uptimes and doesn't serve this data we could consider removing it. I'm not sure how this behaves with our codec/if it's safe to do though @StephenButtolph .